var/spack/repos/builtin/packages/ncbi-toolkit/package.py

```diff
@@ -3,16 +3,23 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+from glob import glob
+
 from spack import *
 
 
 class NcbiToolkit(AutotoolsPackage):
     """NCBI C++ Toolkit"""
 
     homepage = "https://www.ncbi.nlm.nih.gov/IEB/ToolBox/CPP_DOC/"
-    url      = "ftp://ftp.ncbi.nih.gov/toolbox/ncbi_tools++/CURRENT/ncbi_cxx--21_0_0.tar.gz"
+    url      = "ftp://ftp.ncbi.nih.gov/toolbox/ncbi_tools++/CURRENT/ncbi_cxx--22_0_0.tar.gz"
+
+    version('22_0_0', 'e352d25b24c3a2d087c2cf3cedf6ea95')
```

Comment:
It looks like this version can also be found in the ARCHIVE. Should we use that URL by default so the current one doesn't break when a new release comes out?

Reply:
Applied to 0a094fe

I confirmed that downloading 22_0_0 uses ARCHIVE and trying a new unreleased version automagically looks in CURRENT:

```console
$ spack fetch ncbi-toolkit@22_0_0
==> Fetching ftp://ftp.ncbi.nih.gov/toolbox/ncbi_tools++/ARCHIVE/2019/Mar_28_2019/ncbi_cxx--22_0_0.tar.gz
############################################################################################################################################ 100.0%
```

```console
$ spack fetch ncbi-toolkit@23_0_0
==> Warning: There is no checksum on file to fetch ncbi-toolkit@23_0_0 safely.
==>   Fetch anyway? [y/N] y
==> Fetching with no checksum.
  Add a checksum or use --no-checksum to skip this check.
==> Fetching ftp://ftp.ncbi.nih.gov/toolbox/ncbi_tools++/CURRENT/ncbi_cxx--23_0_0.tar.gz
#=O#-   #     #                                                                                                                                   
curl: (78) RETR response: 550
==> Fetching from ftp://ftp.ncbi.nih.gov/toolbox/ncbi_tools++/CURRENT/ncbi_cxx--23_0_0.tar.gz failed.
==> Error: All fetchers failed for ncbi-toolkit-23_0_0-p4pm3qovnyt6hrld2sm62rjws6hk73sw
```